src/rosdiscover/recover/symbolic.py

```diff
@@ -842,6 +922,34 @@ def build(cls, entrypoint: str, functions: t.Iterable[SymbolicFunction]) -> Symb
             name_to_function[entrypoint] = SymbolicFunction.empty(entrypoint)
         return SymbolicProgram(entrypoint, name_to_function)
 
+    def func_of_stmt(self, stmt: SymbolicStatement) -> SymbolicFunction:
+        for func in self.functions.values():
+            if stmt in func.body:
```

Comment:
will this work for functions that contain compound statements?

Reply:
good question. Probably not. I'll verify

Reply:
Fixed